translate/deeplx/deeplx.go

```diff
@@ -0,0 +1,99 @@
+package deeplx
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/metatube-community/metatube-sdk-go/common/fetch"
+	"github.com/metatube-community/metatube-sdk-go/translate"
+)
+
+var _ translate.Translator = (*DeepLX)(nil)
+
+type DeepLX struct {
+	APIKey  string `json:"deepl-api-key"`
+	BaseURL string `json:"base-url"`
+}
+
+func (dpl *DeepLX) Translate(q, source, target string) (result string, err error) {
+	if dpl.BaseURL == "" {
+		return "", translate.ErrTranslator
+	}
+
+	if source == "" {
+		source = "auto"
+	}
+
+	reqBody := map[string]string{
+		"text":        q,
+		"source_lang": parseToDeeplSupportedLanguage(source),
+		"target_lang": parseToDeeplSupportedLanguage(target),
+	}
+
+	opts := []fetch.Option{
+		fetch.WithRaiseForStatus(true),
+		fetch.WithHeader("Content-Type", "application/json"),
+		fetch.WithHeader("Accept", "*/*"),
+		fetch.WithHeader("User-Agent", "MetaTube/1.0.0"),
+		fetch.WithHeader("Connection", "keep-alive"),
+	}
+
+	if dpl.APIKey != "" {
+		opts = append(opts,
+			fetch.WithHeader("Authorization", "DeepL-Auth-Key "+dpl.APIKey),
+		)
+	}
+
+	var resp *http.Response
+	if resp, err = fetch.Post(dpl.BaseURL, fetch.WithJSONBody(reqBody), opts...); err != nil {
+		return "", fmt.Errorf("request failed: %v", err)
+	}
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to read response body: %v", err)
+	}
+	fmt.Printf("Response: %s\n", string(respBody))
```

Comment:
I think `fmt.Printf` should not be used as a logger. could just remove this line